NewForce-Cohort-5/tabloidfullstack-actual-spring
# Nss 3 viewall

Trello ticket #3 - Viewing all posts

The code in this pull request allows the user to click on the "Posts" in the navigation bar of the website and view all approved posts within the Posts page.
Once the "Posts" navigation bar item is clicked, a list of all posts are shown. Each post shows the title and author of the post.
Each post contains the approved posts and has a publication date that is in the past.
The list of posts are in the order of publication date with the most recent on the top.

Backend - These files were created:
PostController.cs, Post.cs, IPostRepository.cs, PostRepository.cs
Startup.cs file was updated based on the interface and repository files.

Frontend - These files were created:
Post.js, PostDetail.js, PostForm.js, PostList.js, PostProvider.js
ApplicationViews.js file was updated with the PostProvider tags and route path to PostList method.
Header.js file was updated with the NavLink tag to include "Posts" in the navigation bar.

Fixes # (issue)
No issues were fixed, new features were added to view all approved posts.

# Testing Instructions
Run the server side in Visual Studio (Purple icon)
Run "npm start" command in Gitbash within this directory:
~/workspace/Back-End/GroupProject/FullStackTabloid/tabloidfullstack-actual-spring/Tabloid/client
Swagger API and the webpage will show up on separate tabs within your browser.
In the webpage, log is using "foo@bar.com" as username and "test" as password.
On the main page click on the "Posts" tab.
You should be able to see all approved posts.


# Checklist:

- [ X ] My code follows the style guidelines of this project
- [ X ] I have performed a self-review of my own code
- [ X ] My changes generate no new warnings or errors
